assets/js/search.js

```diff
@@ -164,7 +153,7 @@
               searchContent = entry.title + entry.category + entry.content + entry.tags;
               break;
             case searchEntry.type === 'guideEntries' && type === 'tagSearch':
```

Comment:
Not related to this PR, but do we have a Jira card to clean this up? I mentioned a couple times to @hngerebara that this super-hardcoded stuff means this search code is neither generic/reusable, nor particularly easy to read/maintain. Happy to chat about how to improve.

Reply:
I did some very small improvements on the parts that I touched or little things that would not alter the logic (such as this replacement of `.map` for `.forEach`), as I was fixed on fixing things asap. But yes, a better refactor would be ideal. Another needed refactor is splitting `global.less` into individual smaller files like we do with the js. I think hope created a card for that one, but I'm gonna ask her if there's a card for refactoring this too. Although to be completely honest I'm not sure if it's worth a lot of effort soon given we are migrating to a dynamic app and we can just make better choices and use an external lib or just plain react state management for things like this. I guess we should make some common sense approach depending how far we get from migrating to the dynamic app.

Reply:
+1 on figuring out if it makes sense to touch this further. Definitely not a high priority.